### ISSUE TITLE
Linear regression (R2) fails when boostrapping on very few data points

### ISSUE THREAD

**Issue by IAlibay** [OpenFreeEnergy/cinnabar#124]
## What happens?

1. When plotting dGs, the statistics calculations attempt to generate 1000 boostraps (with replacements) of the input dataset.

2. If one of the boostraps yields the same number for all the values, the linear regression fails (because you can analyze a flat line).

3. If you have 4 data points, this means that there is a 0.39% chance that you will pick the same number 4 times in a row. I.e. ~ 3 boostraps will have the same value for all the data points.

## Should this even happen?

Probably not, calculating correlation metrics on so few data points is probably not sensible. It may be that the answer here is to just not report correlation metrics when there are fewer than a set number of data points (at least 10?).

**Comment by jthorton** [comment]
In some of the recent changes to the stats functions, this seems to have been partially fixed; the plots will now generate with the CI set to nan, but is this what we want? On one hand, it tells the user it ran bootstrapping, and it failed but it doesn't look great.

<img width="340" height="427" alt="Image" src="https://github.com/user-attachments/assets/24e3057a-a0ae-4446-85c0-36764092959b" />

**Comment by IAlibay** [comment]
I personally think it'd be good for us to be opiniated and just say "hey if you have fewer N ligands, you really really shouldn't do this".

My suggestion is to set it to the same number we used as a cutoff in the industry benchmark studies.

**Comment by jthorton** [comment]
> My suggestion is to set it to the same number we used as a cutoff in the industry benchmark studies.

I don't think we had a cutoff it looks like we tried to calculate it, and if it failed, just did it without https://github.com/OpenFreeEnergy/IndustryBenchmarks2024/blob/main/industry_benchmarks/analysis/overall_dg_analysis.ipynb. We could do the same and just not report values with a nan in the CI interval in the plots or expose this as yet another option like `hide_nan_ci`?
